### PR TITLE
Add jOOQ formula

### PR DESCRIPTION
**Pull request by benmccann**
Closes https://github.com/micronaut-projects/micronaut-starter/issues/151

@lightoze is there more we should add to this in a follow up PR? In any case, this seems like a good start to me

I have tests fail with this branch locally, but only ones that are failing on `master`

### CODE REVIEW

File: starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        then:
+        template.contains('implementation("io.micronaut.configuration:micronaut-jooq")')
+
+        where:
+        language << Language.values().toList()
+    }
+
+    @Unroll
+    void 'test maven cassandra feature for language=#language'() {
```
Comment (jameskleeh) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy:25]: Fix test description

File: starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import io.micronaut.starter.application.ApplicationType
+import io.micronaut.starter.feature.build.gradle.templates.buildGradle
+import io.micronaut.starter.feature.build.maven.templates.pom
+import io.micronaut.starter.options.Language
+import spock.lang.Unroll
+
+class JooqSpec extends BeanContextSpec {
+
+    @Unroll
+    void 'test gradle cassandra feature for language=#language'() {
```
Comment (jameskleeh) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy:13]: Fix test description

File: starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy (hunk 1 above)
Comment (benmccann) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy]: done

File: starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy (hunk 2 above)
Comment (benmccann) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/database/JooqSpec.groovy]: done. thanks for catching those

**Comment by graemerocher** [comment]
I think like the Hibernate feature this feature probably needs to apply the JDBC configuration and add H2 by default so it works out of the box otherwise the user will have to add configuration and manually pick a JDBC driver. See:

https://github.com/micronaut-projects/micronaut-starter/blob/master/starter-core/src/main/java/io/micronaut/starter/feature/database/HibernateJpa.java#L52-L62

**Comment by benmccann** [comment]
Ok. I've added H2. It might be nice if there were the ability to specify options for features. I use Postgres so it's a little weird to add the H2 dependency to my project. We could have a dropdown that defaulted to H2, but let users choose Postgres or MySQL instead. It could maybe be under an advanced toggle if we want to keep things simple by default.

**Comment by jameskleeh** [comment]
@benmccann The H2/Mysql/etc could easily be implemented. `--features mysql-driver` for example. You can submit an issue to track the request for that support

**Pull request merged**